pkg/manager/task/controller.go

```diff
@@ -47,8 +50,27 @@ func (c *Controller) updateTaskAttributes(taskId string, attributes map[string]i
 	return err
 }
 
+var mutex sync.Mutex
+
+func (c *Controller) GetTaskLength() uint32 {
+	// TODO: from global config
+	return constants.TaskLength
+}
+
+func (c *Controller) IsRunningExceed() bool {
+	mutex.Lock()
+	defer mutex.Unlock()
+	count := c.runningCount
+	return count > c.GetTaskLength()
+}
+
 func (c *Controller) ExtractTasks() {
 	for {
+		if c.IsRunningExceed() {
+			logger.Errorf("Sleep 10s, running task count exceed [%d/%d]", c.runningCount, c.GetTaskLength())
+			time.Sleep(10 * time.Second)
+			continue
+		}
 		taskId, err := c.queue.Dequeue()
```

Comment:
I see in dequeue, If the queue is empty, Dequeue blocks until elements are available.

Reply:
Sorry, I know. Here is needed.